txscript: Add pay to PKH and SH tests

JoeGruffins opened the pull request.

teknico commented on decred/tests/unit/dcr/test_txscript.py:
```diff
@@ -4448,3 +4448,147 @@ def test_as_small_int():
+def test_pay_to_script_hash_script():
+    """
+    name (str): Short description of the test.
+    scriptHash (ByteArray): The script hash to pay to.
```
`name` is not there, and `scriptHash` should be `res`.

teknico commented on decred/tests/unit/dcr/test_txscript.py:
```diff
@@ -4448,3 +4448,147 @@ def test_as_small_int():
+def test_pay_to_pubkey_script():
+    """
+    name (str): Short description of the test.
+    serializedPubkey (ByteArray): The pubkey to pay to.
```
`name` is not there, and `serializedPubkey` should be `res`.

teknico commented on decred/tests/unit/dcr/test_txscript.py:
```diff
@@ -4448,3 +4448,147 @@ def test_as_small_int():
+    stakeCode (int): An op code for the type of stake submission transaction.
+        SSTX or SSTXCHANGE.
+    want (ByteArray): The script that pays to the pubkey hash of the address.
+    """
```
Using a multi-line string like this is unusual. It looks like a docstring but it's not in the docstring place. Maybe turn it into a multi-line comment?

JoeGruffins replied:
Been doing this for all the tests though, to document the named variables. By multi-line comment you mean to use `#` I guess? Will change for this test if it's confusing. Some of the txscript tests still used unittest too, so I guess someone could change all the other instances while updating, if that were to happen.

teknico commented on decred/tests/unit/dcr/test_txscript.py:
```diff
@@ -4448,3 +4448,147 @@ def test_as_small_int():
+    stakeCode (int): An op code for the type of stake submission transaction.
+        SSTX or SSTXCHANGE.
+    want (ByteArray): The script that pays to the pubkey hash of the address.
+    """
```
Same as above.

teknico commented on decred/tests/unit/dcr/test_txscript.py:
```diff
@@ -4448,3 +4448,147 @@ def test_as_small_int():
+def test_pay_to_ssrtx_pubkey_hash_direct():
+    """
+    name (str): Short description of the test.
+    pkh (ByteArray): The pubkey hash bytes to pay to.
```
`name` is not there, and `pkh` should be `res`.

teknico commented on decred/tests/unit/dcr/test_txscript.py:
```diff
@@ -4448,3 +4448,147 @@ def test_as_small_int():
+def test_pay_to_ssrtx_script_hash_direct():
+    """
+    name (str): Short description of the test.
+    pkh (ByteArray): The script hash bytes to pay to.
```
`name` is not there, and `pkh` should be `res`.

JoeGruffins commented:
Sorry about the review changes being part of the initial comment.  I mistakenly made the changes in the middle of a rebase.

I removed the docstrings from the short tests. I think the test names are descriptive enough to give enough information about what's going on. I changed that other docstrings into a muliti-line comment, if I understood correctly.

Will work on PayToStake validation in another pr.

JoeGruffins commented:
follow up in #150
